py_cui/control_widgets/slider.py

```diff
@@ -4,12 +4,109 @@
 
 
 class SliderImplementation(py_cui.ui.UIImplementation):
-    pass
+
+    def set_bar_char(self,char):
+        self._bar_char = char
+
+    def update_slider_value(self, direction):
+        """ set the value of the slider - increment decrement
+        """
+        # direction , 1 raise value, -1 lower value
+        self._cur_val += (direction * self._step)
+        if (self._cur_val <= self._min_val):
+            self._cur_val = self._min_val
+        if (self._cur_val >= self._max_val):
+            self._cur_val = self._max_val
+
+        return self._cur_val
+
+    def get_slider_value(self):
+        """return current slider value
+        """
+        return self._cur_val
+
+    def set_slider_step(self,step):
+        """change step value
+        """
+        self._step = step
 
 
 class SliderWidget(py_cui.widgets.Widget, SliderImplementation):
-    pass
+    """Widget for a Slider
+    """
+
+    """
+    Attributes
+    ----------
+    _min_val : int
+        Lowest value of the slider
+    _max_val: int
+        Highest value of the slider
+    _step : int
+        Increment from low to high value
+    _cur_val:
+        Current value of the slider
+
+    """
+    _bar_char = '#'
+
+    def __init__(self, id, title, grid, row, column, row_span, column_span,
+                 padx, pady, logger, min_val, max_val, step, init_val):
+
+        super().__init__(id, title, grid, row, column,
+                         row_span, column_span, padx, pady, logger,
+                         selectable=True)
+
+        self._min_val = min_val
```

Comment:
These attributes should be moved to the Implementation class, along with the bar_char attribute, since they don't depend on whether they are contained in a widget or popup. Also, you will need to call both the `Widget` superclass initializer and the `SliderImplementation` intializer, to make sure the variables are all correctly initialized.

Reply:
done, I've put a little comment in draw method. suggestions?